### ISSUE TITLE
How to override the highlighting of the @include group in a specific language?

### ISSUE THREAD

**Issue by ahmedelgabri** [nvim-treesitter/nvim-treesitter#1346]
I'm trying to override import statement highlights in JS/TS/JSX/TSX but I can't seem to be able to override `import`, `from` & `as` as they are part of the [`@include`](https://github.com/nvim-treesitter/nvim-treesitter/blob/972f70956ac8635ab2f6d7ed0b24e5cd159d7a04/queries/ecma/highlights.scm#L190-L194) group.

This is the max I could do with these queries (_I'm still figuring out how queries work, so they might not be correct_)

```query
(import_statement) @comment
; (named_imports) @comment
(import_statement (import_clause) @comment)
(import_statement source: (string) @comment)
(import_statement (import_clause (namespace_import) @comment))
(import_clause (identifier) @comment_info)
(namespace_import (identifier) @comment_info)
(import_clause (named_imports) @comment_info)
(import_clause (named_imports (import_specifier)) @comment)
(import_specifier name: (identifier) @comment_info)
```

<img width="502" alt="Screenshot 2021-05-29 at 21 28 49" src="https://user-images.githubusercontent.com/63876/120082769-0fbbfa80-c0c5-11eb-94ba-9fc2ed270d09.png">

If I add these to `custom_captures` 

```lua
    custom_captures = {
      ["include"] = "",
      ["punctuation.bracket"] = ""
    }
```

I can reach this

<img width="481" alt="Screenshot 2021-05-29 at 21 28 19" src="https://user-images.githubusercontent.com/63876/120082818-5b6ea400-c0c5-11eb-85fb-ac9673d8dcce.png">

This is closer to what I want, but as far as I understand this will override `@include` & `@punctuation.bracket` for all languages & I don't want that. I tried `tsx.include`, `ts.include`, `js.include`, `jsx.include`, even `ecma.include`, none worked. (_Same applied to `punctuation.bracket`. It will be even better if I can target only the punctuation inside of the import only_)

So to recap, this is what I'm trying to do:

- I want to override `@include` group in JavaScript/TypeScript (and jsx/tsx)
- if needed, I want to override `@punctuation.bracket` only inside an `import`

**Comment by theHamsta** [comment]
> I want to override @include group in JavaScript/TypeScript (and jsx/tsx)

You can try to install the nvim-treesitter/playground and use TSHighlightCapturesUnderCursor. It will show you that a mapping in two steps is performed. `@include` -> `jsTSInclude` which is  default linked to TSInclude. If you want to  change the highlight just for JS you can set you custom highlight for TSInclude and link it to a different highlight group.

If you want to give the dots in a include in a specific language a different highlight, use `:TSEditUserQuery highlight` to write your own query and use `@MyHighlightGroup` directly

**Comment by ahmedelgabri** [comment]
> You can try to install the nvim-treesitter/playground and use TSHighlightCapturesUnderCursor. It will show you that a mapping in two steps is performed. `@include` -> `jsTSInclude` which is default linked to TSInclude. If you want to change the highlight just for JS you can set you custom highlight for jsTSInclude and link it to a different highlight group.

This one works, thanks



> If you want to give the dots in a include in a specific language a different highlight, use `:TSEditUserQuery highlight` to write your own query and use `@MyHighlightGroup` directly

I don't quite understand what you mean here, can you elaborate a bit more?

**Comment by theHamsta** [comment]
You have to write your own query when you want to capture of a specific query. You should see the syntax tree in playground. Press the `a` key to also see anonymous nodes like parens. This should help you to write the query you want to have.  I just wanted to say that you can also use `@Comment` instead of `@comment` then it will use the highlight group directly without any mapping.

**Comment by ahmedelgabri** [comment]
Thanks @theHamsta
